pkg/controller/inferenceservice/reconcilers/knative/service_reconciler.go

```diff
@@ -84,51 +67,47 @@ func (r *ServiceReconciler) reconcileCanary(isvc *v1alpha2.InferenceService) err
 }
 
 func (r *ServiceReconciler) reconcileComponent(isvc *v1alpha2.InferenceService, component constants.InferenceServiceComponent, isCanary bool) error {
+	endpointSpec := &isvc.Spec.Default
+	serviceName := constants.DefaultServiceName(isvc.Name, component)
+	propagateStatus := isvc.Status.PropagateDefaultStatus
 	if isCanary {
-		if isvc.Spec.Canary == nil {
-			if err := r.finalizeCanaryService(isvc, component); err != nil {
-				return err
-			}
-			isvc.Status.PropagateCanaryStatus(component, nil)
-			return nil
+		endpointSpec = isvc.Spec.Canary
+		serviceName = constants.CanaryServiceName(isvc.Name, component)
+		propagateStatus = isvc.Status.PropagateCanaryStatus
+	}
+	var service *knservingv1alpha1.Service = nil
+	var err error = nil
```

Comment:
Cleaner to just use the error inline, as with the rest of the codebase.

Reply:
yes, we'd better use error inline. but since `service, err :=` will take `service` as a new var, too. Here I think we have to keep it as it is now